python-scripts/get_timers.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+
+# Prints information about some of the Xbox timers
+
+from xbox import *
+
+import time
+
+NV_PRAMDAC = 0x680000
+NV_PRAMDAC_NVPLL_COEFF = 0x500
+NV_PRAMDAC_NVPLL_COEFF_MDIV = 0x000000FF
+NV_PRAMDAC_NVPLL_COEFF_NDIV = 0x0000FF00
+NV_PRAMDAC_NVPLL_COEFF_PDIV = 0x00070000
+
+NV2A_CRYSTAL_FREQ = 13500000 # Hz
+
+NV_PTIMER = 0x9000
+NV_PTIMER_TIME_0 = 0x400
+NV_PTIMER_TIME_1 = 0x410
+
+if __name__ == "__main__":
+
+  while(True):
+
+    KeTickCount = memory.read_u32(ke.KeTickCount())
+    KeTickCountInSeconds = KeTickCount / 1000
+
+    GPUTimer0 = nv2a.read_u32(NV_PTIMER + NV_PTIMER_TIME_0)
+    GPUTimer1 = nv2a.read_u32(NV_PTIMER + NV_PTIMER_TIME_1)
+    GPUTimer0dec = (GPUTimer0 >> 5) & 0x7FFFFFF
+    GPUTimer1dec = (GPUTimer1 & 0x1FFFFFFF) << 27
+    GPUTimer = GPUTimer1dec | GPUTimer0dec
+    
+    GPUNumerator = nv2a.read_u32(NV_PTIMER + 0x200)
+    GPUDenominator = nv2a.read_u32(NV_PTIMER + 0x210)
+
+    nvpll_coeff = nv2a.read_u32(NV_PRAMDAC + NV_PRAMDAC_NVPLL_COEFF)
+    mdiv = nvpll_coeff & NV_PRAMDAC_NVPLL_COEFF_MDIV
+    ndiv = (nvpll_coeff & NV_PRAMDAC_NVPLL_COEFF_NDIV) >> 8
+    pdiv = (nvpll_coeff & NV_PRAMDAC_NVPLL_COEFF_PDIV) >> 16
+    GPUClockrate = (NV2A_CRYSTAL_FREQ * ndiv) / (1 << pdiv) / mdiv
+    GPUTimerInSeconds = GPUTimer / (GPUClockrate * (GPUNumerator / GPUDenominator))
+
+    print("GPUClockrate = (NV2A_CRYSTAL_FREQ * ndiv) / (1 << pdiv) / mdiv")
+    print("GPUClockrate = (" + str(NV2A_CRYSTAL_FREQ)+" * " + str(ndiv) + ") / (1 << " + str(pdiv) + ") / " + str(mdiv) + ")")
```

Comment:
Tested this on hardware and had some issues. Added / Replaced this code after `GPUClockrate = ...`:

```python
    GPUTimerClockrate = GPUClockrate / (GPUNumerator / GPUDenominator)
    GPUTimerInSeconds = GPUTimer / GPUTimerClockrate

    print("GPU Clockrate: (NV2A_CRYSTAL_FREQ * ndiv) / (1 << pdiv) / mdiv")
    print("GPU Clockrate: (" + str(NV2A_CRYSTAL_FREQ)+" * " + str(ndiv) + ") / (1 << " + str(pdiv) + ") / " + str(mdiv) + " = " + str(GPUClockrate) + " Hz")
    print("GPU Timer clockrate: " + str(GPUClockrate) + " / (" + str(GPUNumerator) + " / " + str(GPUDenominator) + ") = " + str(GPUTimerClockrate) + " Hz")
```

- Also prints the timers clockrate
- Removed unmatched ')' in the GPUClockrate line
- Clock divider for `GPUTimerInSeconds` was inversed (Fixed in `GPUTimerClockrate`)

Note that the timers still drift considerably for me [about 1s per 10 seconds]. This might simply be inaccuracy in hardware, but it might also show an issue with my rounding code.

We should have an issue which improves this script to make it easier to measure the *actual* clockrate so we can fix our formulas more easily. However, this is a follow-up PR.

Current PR is fine with the changes I propose here.